Implementation of the user authorization module

Knyazev-AI opened the pull request:
1. Перемещена логика переключения на GIDSignIn-экран в роутер модуля авторизации;
2. Приведение UI элементов модуля авторизации к соответствию дизайн-макету;
3. Изменение размера изображения логотипа на storyboard, а также в модулях Splash, авторизации для приведения к соответствию дизайн-макета.

Commit:
Refactor: move GIDSignIn transition logic to authorization module router

Commit:
Refactoring: authorization module UI

Curshet committed:
Localization: change default app localization

Commit:
Refactoring: VM authorization to transfer data to the View

Commit:
Add: GoogleSignIn service to the container

Curshet commented on MyBook/Modules/Authorization/Builder/AuthorizationBuilder.swift:
```diff
@@ -22,41 +22,42 @@ private extension AuthorizationBuilder {
     func screenRouter() {
         let window = injector.getObject(from: .application, type: UIWindow.self)
-        let router = AuthorizationRouter(window: window)
+        let googleService = injector.getObject(from: .application, type: GIDSignIn.self)
```
Все **Optinal Type** объекты в builder разворачиваются через **guard**.

Curshet commented on MyBook/Modules/Authorization/Builder/AuthorizationBuilder.swift:
```diff
@@ -22,41 +22,42 @@ private extension AuthorizationBuilder {
-    func view(_ label: UILabel, _ loginButton: GIDSignInButton) -> AuthorizationViewProtocol {
-        AuthorizationView(label: label, loginButton: loginButton)
+
+    func logoImageView() -> UIImageView {
```
В builder объекты, которые потенциально могут быть созданые заново и использованы в других объектах (кнопки, лейблы и пр.) желательно именовать в общем их смысле - label, imageView, button (не относится к googleButton, потому как это спец. кнопка).

Curshet commented on MyBook/Modules/Authorization/Router/AuthorizationRouter.swift:
```diff
@@ -1,12 +1,16 @@
 import Foundation
-import UIKit
```
Почему удалён импорт UIKit, когда его объекты используются в классе?

Curshet commented on MyBook/Modules/Authorization/Router/AuthorizationRouter.swift:
```diff
@@ -1,12 +1,16 @@
+    var action: ((GIDSignInResult?, Error?) -> Void)?
+    
     private weak var viewController: UIViewController?
+    private weak var googleService: GIDSignIn?
```
Делать сервис (синглтон) с доступом по слабой ссылке нет смысла.

Curshet commented on MyBook/Modules/Authorization/Router/AuthorizationRouter.swift:
```diff
@@ -39,12 +43,19 @@ private extension AuthorizationRouter {
-
+    
+    func startGoogleSignIn() {
+        guard let viewController else { return }
```
После **guard** необходим пробел.

Curshet commented on MyBook/Modules/Authorization/Services/AuthorizationUserLogin/AuthorizationUserLogin.swift:
```diff
@@ -0,0 +1,72 @@
+        }
+        
+        getUserToken(result)
+        
```
Лишний пробел.

Curshet commented on MyBook/Modules/Authorization/Services/AuthorizationUserLogin/AuthorizationUserLogin.swift:
```diff
@@ -0,0 +1,72 @@
+            let message = "User succefully logged in with token: \(token), id: \(userID)"
+            self?.action?(message)
+        }
+        
```
Лишний пробел.

Curshet commented on MyBook/Modules/Authorization/Services/AuthorizationUserLogin/AuthorizationUserLoginProtocol.swift:
```diff
@@ -0,0 +1,7 @@
+import UIKit
```
UIKit здесь используется?

Curshet commented on MyBook/Modules/Authorization/View/AuthorizationView.swift:
```diff
@@ -1,15 +1,18 @@
 import Foundation
```
Почему нет импорта UIKit?

Curshet replied:
Исключить импорт Foundation.

Curshet commented on MyBook/Modules/Authorization/View/AuthorizationView.swift:
```diff
@@ -1,15 +1,18 @@
 import Foundation
 import GoogleSignIn
```
Нужен ли импорт GoogleSignIn?

Curshet commented on MyBook/Modules/Authorization/View/AuthorizationView.swift:
```diff
@@ -19,60 +22,91 @@ class AuthorizationView: UIView, AuthorizationViewProtocol {
+            case .imageLogo(let image):
+                logoImageView.image = image
+            
+            case .message(let message):
```
Какой полезный функционал у данного case?

Curshet commented on MyBook/Modules/Authorization/ViewController/AuthorizationViewController.swift:
```diff
@@ -21,14 +21,44 @@ class AuthorizationViewController: UIViewController {
     
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupInitialViewModelEvents()
```
Идея объединить ряд логики в один метод - хорошая. Но вместо обхединения логики нужно использовать 1 вызов VM, который сделает то, что нужно, а не слать подряд несколько событий.

Curshet commented on MyBook/Modules/Authorization/ViewController/AuthorizationViewController.swift:
```diff
@@ -21,14 +21,44 @@ class AuthorizationViewController: UIViewController {
     }
-
+    
+    func handleViewModelEvent(_ event: AuthorizationViewInternalEvent) {
```
Этого метода в VC быть не должно. У View входными событиями будут выходные события VM, выходными - входные события VM. Таким образом в VC дополнительно обрабатывать ничего будет не нужно.

Curshet commented on MyBook/Modules/Authorization/ViewModel/AuthorizationViewModel.swift:
```diff
@@ -13,14 +14,52 @@ class AuthorizationViewModel: AuthorizationViewModelProtocol {
+private extension AuthorizationViewModel {
+    
+    func setupObservers() {
+        
```
Лишний пробел.

Curshet commented on MyBook/Modules/Authorization/ViewModel/AuthorizationViewModel.swift:
```diff
@@ -13,14 +14,52 @@ class AuthorizationViewModel: AuthorizationViewModelProtocol {
-            self?.action?($0)
+            self?.action?(.message($0))
         }
+        
```
Лишний пробел.

Curshet commented on MyBook/Resources/Localization/en.lproj/Localizable.strings:
```diff
@@ -1 +1 @@
-"initialGreeting" = "Для входа используйте учётную запись Google";
+"initialGreeting" = "Use your Google account to log in";
```
Строки должны начинаться с заглавной буквы.

Curshet commented on MyBook/Modules/Authorization/Router/AuthorizationRouter.swift:
```diff
@@ -1,12 +1,16 @@
 
 class AuthorizationRouter: AuthorizationRouterProtocol {
     
+    var action: ((GIDSignInResult?, Error?) -> Void)?
```
У роутера должно быть несколько выходных событий. К примеру: успешная авторизация, неуспешая авторизация, выход из модуля. За этими событиями и должна наблюдать соотв. VM и AppCoordinator.

Curshet reviewed (commented):
**1.** Стараться использовать формат **svg** изображения вместо **png**.
**2.** Имена лок. строк пишутся с заглавной буквы.
**3.** ExtensionsUIColor именуем как -> UIColorExtensions.
**4.** Не реализован функционал выхода из модуля авторизации (хотя бы на начальном этапе).
**5.** Не понятен способ проверки статуса авторизации пользователя на этапе запуска приложения.
**6** AuthorizationRouter не должен хранить в себе сервис GIDSignIn. Необходимо избавиться от ссылки на него.
**7.** Лейбл статуса PR должен быть 1, если была реализация фичи. Не нужно указывать дополнительно рефакторинг, так как обычно это выполняется отдельной задачей, но если он всё же был, то явно в меньшей степени чес реализация фичи. Поэтому лейбл "refactroring" излишен.
**8.** Именование PR абсолютно не информативное. Это рефакторинг или фича? "UI refactoring" -> рефакторинг какого UI? Задача ставилась как "Реализация модуля авторизации пользователя". Если ветка названа хорошо, то PR - нет. См. пример предыдущих PR и замечаний к их названиям, если таковые были.

Curshet commented on MyBook/Modules/Authorization/View/AuthorizationView.swift:
```diff
@@ -19,60 +22,91 @@ class AuthorizationView: UIView, AuthorizationViewProtocol {
-    func setupLoginButton() {
-        loginButton.colorScheme = .dark
-        loginButton.style = .wide
+    func setupLoginButton(_ image: UIImage?) {
```
Подумать над тем, как обеспечить View необходимыми объектами с помощью 1 метода, а не 3-мя различными.

Commit:
Add: calling the user authorization status verification method

Commit:
Refactoring: AuthorizatioVC does not handle AuthorisationVM events

Commit:
Refactoring: remove GIDSignIn reference from AuthorizationRouter

Commit:
Refactoring: move the AuthorizationUserLogin logic to AppUserLogin

Commit:
Add: logo image & login buttons in svg format

Commit:
Refactoring: the name of the UIColor extension

Commit:
Refactoring: names of localized strings

Commit:
Refactoring: events in the authorization module

Устранены следующие замечания к PR9:
1. разворачивание optinal type в AuthorizationBuilder выполнено через guard;
2. импорт библиотеки UIKit в файлах с элементами UI;
3. устранены замечания по код-стайлу (в т.ч. добавлены/удалены пробелы, изменены имена методов);
4. удаление импорта GoogleSignIn из AuthorizationView;
5. замена метода «setupInitialViewModelEvents» AutorizationVC на событие AutorizationVM;
6. для обеспечения AutorizationV необходимыми объектами с помощью одного метода, реализован case с передачей 3 файлов

Commit:
Add: logo image in png format

Commit:
Add: interface for getting GoogleService

Commit:
Refactoring: AuthorizationRouter events

Дополнительно:
- вывод в консоль сообщений о пользователе перемещен в AppUserLogin;
- добавлена структура передаваемых в AuthorizationView данных

Curshet commented on Classes/UIColorExtensions.swift:
```diff
@@ -1,9 +1,7 @@
 import UIKit
 
 extension UIColor {
-    
```
Необходимо вернуть пробелы.

Curshet commented on MyBook/Modules/Application/Coordinator/AppCoordinator.swift:
```diff
@@ -15,6 +15,7 @@ class AppCoordinator: AppCoordinatorProtocol {
     
     func start() -> Bool {
         startSplashModule()
+        interactor.sendEvent()
```
Данный метод здесь находиться не должен. Запрос авторизации необходимо выполнять на выходе из модуля Splash, после чего следует логика выбора загрузки следующего модуля - Authorization либо Menu. Логика выбора загрузки следующего модуля отсутствует (см. метод **userAuthorizationHandler**).

Curshet commented on MyBook/Modules/Authorization/Builder/AuthorizationBuilder.swift:
```diff
@@ -83,5 +84,20 @@ extension AuthorizationBuilder {
+
+}
+
+// MARK: Protocol AuthorizationBuilderVCProtocol
```
**// MARK: AuthorizationBuilderProtocol**

В данном случае подпись под протокол необходимо делать именно в extension.

Curshet commented on MyBook/Modules/Authorization/Router/AuthorizationRouter.swift:
```diff
@@ -1,13 +1,18 @@
+    var action: ((AuthorizationRouterExternalEvent) -> Void)?
+    
     private weak var viewController: UIViewController?
     private weak var window: UIWindow?
```
UIWindow также необходимо получать из builder.

Curshet commented on MyBook/Modules/Authorization/Router/AuthorizationRouter.swift:
```diff
@@ -39,12 +47,40 @@ private extension AuthorizationRouter {
+}
+
+// MARK: - AuthorizationRouterInternalEvent
+enum AuthorizationRouterExternalEvent {
```
От доп. типа необходимо избавиться. В подобных случаях используем "Result".

Curshet commented on MyBook/Modules/Application/Interactor/Services/AppUserLogin/AppUserLogin.swift:
```diff
@@ -11,10 +11,63 @@ class AppUserLogin: AppUserLoginProtocol {
+            let message = "User succefully logged in with token: \(token), id: \(userID)"
+            print(message)
+        }
+        
```
Лишний пробел.

Curshet commented on MyBook/Modules/Application/Interactor/Services/AppUserLogin/AppUserLogin.swift:
```diff
@@ -11,10 +11,63 @@ class AppUserLogin: AppUserLoginProtocol {
         }
     }
     
+    func getUserToken(_ result: GIDSignInResult?) {
```
**1.** Этот же метод должен возвращать токен пользователя в "action". 
**2.** От "AppUserLoginInternalEvent" необходимо избавиться.

Knyazev-AI replied:
Замечание не устранено.
AppUserLoginInternalEvent необходим для 2-ух разных ситуаций:
- case authorization при проверке предыдущей авторизации;
- case userLogin(_ user: GIDGoogleUser?) при попытке настоящей авторизации

Curshet commented on MyBook/Modules/Authorization/ViewModel/AuthorizationViewModel.swift:
```diff
@@ -1,26 +1,78 @@
+        }
+    }
+    
+    func internalEventHadler(_ event: AuthorizationViewModelInternalEvent) {
```
Не совсем удачный нейминг метода. Метод отвечает за обработку событий от роутера и нахывать его необходимо соотв. образом.

Curshet commented on MyBook/Modules/Authorization/ViewModel/AuthorizationViewModel.swift:
```diff
@@ -1,26 +1,78 @@
+private extension AuthorizationViewModel {
+    
+    func setupObservers() {
+        router?.action = { [weak self] event in
```
Аргументы, по возможности, лучше использовать безымянные -> $0.

Curshet commented on MyBook/Modules/Authorization/ViewModel/AuthorizationViewModel.swift:
```diff
@@ -1,26 +1,78 @@
+        let imageLogo = UIImage(named: "Logo")
+        let imageLoginButton = UIImage(named: "LoginButton")
+        let text = NSLocalizedString("InitialGreeting", comment: "")
+        let authorizationViewData = AuthorizationViewData(imageLogo: imageLogo, 
```
Такое форматирование аргументов "столбиком" желательно не использовать.

Curshet commented on MyBook/Modules/Authorization/ViewModel/AuthorizationViewModel.swift:
```diff
@@ -1,26 +1,78 @@
+    func internalEventHadler(_ event: AuthorizationRouterExternalEvent) {
+        switch event {
+            case .failure(let error):
+                print("Error user login with Google: \(error!.localizedDescription)")
```
Force Unwrapping желательно не использовать. Необходимо избавиться.

Curshet commented on MyBook/Modules/Application/Interactor/Services/AppUserLogin/AppUserLogin.swift:
```diff
@@ -11,10 +11,63 @@ class AppUserLogin: AppUserLoginProtocol {
+                return
+            }
+            
+            guard let token = user.idToken?.tokenString else {
```
Достаточно ли здесь проверки только на Optional?

Curshet commented on MyBook/Modules/Application/Interactor/Services/AppUserLogin/AppUserLogin.swift:
```diff
@@ -11,10 +11,63 @@ class AppUserLogin: AppUserLoginProtocol {
+                return
+            }
+            
+            guard let userID = user.userID else {
```
Достаточно ли здесь проверки только на Optional?

Curshet commented on MyBook/Modules/Authorization/ViewModel/AuthorizationViewModel.swift:
```diff
@@ -1,26 +1,78 @@
-            self?.action?($0)
+    func internalEventHadler(_ event: AuthorizationRouterExternalEvent) {
+        switch event {
+            case .failure(let error):
```
Зачем необходим данный case в принципе? Схожий функционал выполняет AppUserLogin (печать ошибки авторизации пользователя в консоль).

Knyazev-AI replied:
Замечание не устранено.
Вывод указанного сообщения происходит в:
- AppUserLogin при проверке предыдущей авторизации;
- AuthorizationViewModel при попытке настоящей авторизации

Commit:
Add: menu module

Commit:
Refactoring: getting a window from splashBuilder

Commit:
Refactoring: getting a window from authorizationBuilder

Commit:
Refactoring: AppBuilder - adding the menu module

Commit:
Refactoring: module switching logic

Curshet commented on MyBook/Modules/Application/Coordinator/AppCoordinator.swift:
```diff
@@ -36,17 +36,19 @@ private extension AppCoordinator {
+            case .avaliable:
+                startMenuModule()
+            }
+        
```
Лишний пробел.

Curshet commented on MyBook/Modules/Application/Coordinator/AppCoordinator.swift:
```diff
@@ -36,17 +36,19 @@ private extension AppCoordinator {
-        }
+            case .avaliable:
+                startMenuModule()
+            }
```
Code Style: закрывающая скобка должна располагаться на одном уровне с началом ключеового слова "switch".

Curshet commented on MyBook/Modules/Application/Coordinator/AppCoordinator.swift:
```diff
@@ -36,17 +36,19 @@ private extension AppCoordinator {
                 break
-        }
+            }
+        
```
Лишний пробел.

Curshet commented on MyBook/Modules/Application/Coordinator/AppCoordinator.swift:
```diff
@@ -36,17 +36,19 @@ private extension AppCoordinator {
             case .pushNotitication(_):
                 break
-        }
+            }
```
Code Style: закрывающая скобка должна располагаться на одном уровне с началом ключеового слова "switch".

Curshet commented on MyBook/Modules/Application/Interactor/AppInteractor.swift:
```diff
@@ -12,23 +11,34 @@ class AppInteractor: AppInteractorProtocol {
+            case .unavaliable:
+                action?(.authorization(.unavaliable))
+                print("User didn't log in")
+            }
```
Code Style: скобка/уровень.

Curshet commented on MyBook/Modules/Application/Interactor/AppInteractor.swift:
```diff
@@ -12,23 +11,34 @@ class AppInteractor: AppInteractorProtocol {
+        switch event {
+            case .avaliable:
+                action?(.authorization(.avaliable))
+                print("User logged in")
```
Насколько вообще здесь нужны какие-либо выводы в консоль, если в избытке информация логируется с AppUserLogin?

Knyazev-AI replied:
Прошу уточнить по замечанию.

Сейчас методы AppUserLogin не выводят информацию о предыдущей авторизации пользователя. Они лишь вызывают соответствующее событие. Переместить вывод указанной информации в AppUserLogin ?

Curshet commented on MyBook/Modules/Application/Interactor/Services/AppUserLogin/AppUserLogin.swift:
```diff
@@ -3,18 +3,108 @@ import GoogleSignIn
     
-    var action: ((GIDGoogleUser?) -> Void)?
+    var action: ((AppInteractorUserStatus) -> Void)?
+    var completeAction: (() -> Void)?
```
Если необходимо отправить внешнее событие из объекта, то нужно пользоваться только свойством "action", в доп. свойствах нет необходимости.

Knyazev-AI replied:
Прошу подсказать по возможности.
При добавлении дополнительного события в enum AppInteractorUserStatus возникает необходимость обработки всех case в разных частях программы, не нуждающихся в выполнении определенных условий при определенном case. 
Возможно ли обрабатывать только одно из условий case ?

Curshet commented on MyBook/Modules/Application/Interactor/Services/AppUserLogin/AppUserLogin.swift:
```diff
@@ -3,18 +3,108 @@ import GoogleSignIn
+}
+
+// MARK: - ResultGetToken
+enum ResultGetToken {
```
В этом типе нет необходимости. Имеется встроенный тип "Result".

Curshet commented on MyBook/Modules/Application/Interactor/Services/AppUserLogin/AppUserLogin.swift:
```diff
@@ -3,18 +3,108 @@ import GoogleSignIn
+        }
+    }
+    
+    func resultGetToken(_ result: ResultGetToken) {
```
Зачем нужен этот метод, если логирование успеха/ошибки можно делать в методе "getUserToken"?

Curshet commented on MyBook/Modules/Application/Interactor/Services/AppUserLogin/AppUserLogin.swift:
```diff
@@ -3,18 +3,108 @@ import GoogleSignIn
         }
     }
     
+    func getUserToken(_ user: GIDGoogleUser?, completion: @escaping (ResultGetToken) -> Void) {
```
Насколько здесь необходим completion? Задача метода проверить токен на валидность и отправить в свойство "action" событие.

Curshet commented on MyBook/Modules/Application/Interactor/Services/AppUserLogin/AppUserLogin.swift:
```diff
@@ -3,18 +3,108 @@ import GoogleSignIn
+                completion(.failure(error))
+                return
+            }
+            let message = "User succefully logged in with token: \(token), id: \(userID)"
```
Добавить пробел.

Curshet commented on MyBook/Modules/Application/Interactor/Services/AppUserLogin/AppUserLogin.swift:
```diff
@@ -3,18 +3,108 @@ import GoogleSignIn
+                case .failure(let error):
+                    self?.action?(.unavaliable)
+                    print(error)
+                }
```
Скобка/уровень.

Curshet commented on MyBook/Modules/Application/Interactor/Services/AppUserLogin/AppUserLogin.swift:
```diff
@@ -3,18 +3,108 @@ import GoogleSignIn
+            
+            case .failure(let error):
+                print(error)
+            }
```
Скобка/уровень.

Curshet commented on MyBook/Modules/Authorization/Builder/AuthorizationBuilder.swift:
```diff
@@ -21,59 +22,58 @@ private extension AuthorizationBuilder {
 }
 
-// MARK: Protocol
+// MARK: Protocol AuthorizationBuilderProtocol
```
**// MARK: AuthorizationBuilderProtocol**

Curshet commented on MyBook/Modules/Authorization/Builder/AuthorizationBuilder.swift:
```diff
@@ -83,5 +83,18 @@ extension AuthorizationBuilder {
+
+}
+
+// MARK: Protocol AuthorizationBuilderRoutProtocol
```
**// MARK: AuthorizationBuilderRoutProtocol**

Curshet commented on MyBook/Modules/Authorization/Builder/AuthorizationBuilderRoutProtocol.swift:
```diff
@@ -0,0 +1,8 @@
+import Foundation
```
Если имеется импорт UIKit, то Foundation уже ни к чему.

Curshet commented on MyBook/Modules/Authorization/Router/AuthorizationRouter.swift:
```diff
@@ -1,13 +1,17 @@
 import Foundation
```
Избавиться от импорта Foundation.

Curshet commented on MyBook/Modules/Authorization/Router/AuthorizationRouter.swift:
```diff
@@ -27,24 +31,40 @@ private extension AuthorizationRouter {
-
+    
+    func startGoogleSignIn() {
+        guard let viewController else { return }
```
**Upd:** пробел всё также также отсутствует.

Curshet commented on MyBook/Modules/Authorization/Router/AuthorizationRouterProtocol.swift:
```diff
@@ -1,5 +1,8 @@
 
 protocol AuthorizationRouterProtocol: AnyObject {
+    var action: ((Result<GIDGoogleUser?, Error>) -> Void)? { get set }
+    var completeAction: (() -> Void)? { get set }
```
Аналогично свойству AppUserLogin -> для внешних событий использовать исключительно свойство "action".

Knyazev-AI replied:
Прошу подсказать по возможности.
При добавлении дополнительного события в enum типа возникает необходимость обработки всех case в разных частях программы, не нуждающихся в выполнении определенных условий при определенном case.
Возможно ли обрабатывать только одно из условий case ?

Curshet commented on MyBook/Modules/Authorization/View/AuthorizationView.swift:
```diff
@@ -19,60 +22,76 @@ class AuthorizationView: UIView, AuthorizationViewProtocol {
+
+// MARK: - AuthorizationViewData
+struct AuthorizationViewData {
+    var imageLogo: UIImage?
```
Почему используется **var**, а не **let**?

Curshet commented on MyBook/Modules/Navigation/Menu/Builder/MenuBuilder.swift:
```diff
@@ -0,0 +1,69 @@
+import Foundation
```
Исключить импорт Foundation.

Curshet commented on MyBook/Modules/Navigation/Menu/Builder/MenuBuilder.swift:
```diff
@@ -0,0 +1,69 @@
+    
+}
+
+// MARK: Protocol MenuBuilderProtocol
```
Избавиться от "Protocol....."

Curshet commented on MyBook/Modules/Navigation/Menu/Builder/MenuBuilder.swift:
```diff
@@ -0,0 +1,69 @@
+    
+}
+
+// MARK: Protocol MenuBuilderRoutProtocol
```
Избавиться от "Protocol....."

Curshet commented on MyBook/Modules/Navigation/Menu/Builder/MenuBuilderRoutProtocol.swift:
```diff
@@ -0,0 +1,6 @@
+import Foundation
```
Избавиться от импорта Foundation.

Curshet commented on MyBook/Modules/Navigation/Menu/Router/MenuRouter.swift:
```diff
@@ -0,0 +1,49 @@
+import Foundation
```
Избавиться от импорта Foundation.

Curshet commented on MyBook/Modules/Navigation/Menu/ViewController/MenuTabBarController.swift:
```diff
@@ -0,0 +1,16 @@
+import Foundation
```
Избавиться от импорта Foundation.

Curshet commented on MyBook/Modules/Splash/Builder/SplashBuilderRoutProtocol.swift:
```diff
@@ -0,0 +1,6 @@
+import Foundation
```
Избавиться от импорта Foundation.

Commit:
Refactoring: logging AppUserLogin methods

Commit:
Refactoring: external AuthorizationRouter events

Curshet committed:
Add: reactive function of authorization router external events

Commit:
Add: external event management service

Commit:
Refactoring: optional argument has been added to the event service

Commit:
Refactoring: an extension has been added to gitignore ".xcbkptlist"

Commit:
Add: Publishers service for sending data and subscribing to them

Commit:
Refactoring: the data type "Optional" is set in the Publishers service

Commit:
Add: publishers service to the Application module

Additionally, the naming of some methods has been fixed

Commit:
Add: publishers service to the Authorization module

Additionally, the naming of some methods has been fixed

Commit:
Add: publishers service to the Menu module

Additionally, the naming of some methods has been fixed

Commit:
Add: publishers service to the Splash module

Additionally, the naming of some methods has been fixed

Commit:
Refactoring: "Optional" type in the publishers service removed

Commit:
Refactoring: naming properties of Application module has been changed

Commit:
Refactoring: naming properties of Authorization module has been changed

Commit:
Refactoring: naming properties of Menu module has been changed

Commit:
Refactoring: naming properties of Splash module has been changed

Commit:
Add: checking "Optional" type when injecting VC

Additionally, the naming of events has been fixed

Commit:
Delete: optionality of the text parameter in setupLabelGreeting

Commit:
Add: checking for nil when injecting ViewController

In addition, the naming of the input parameter has been changed when injecting the ViewController

Commit:
Delete: "self" when checking for optionality

Commit:
Delete: "self" when checking for optionality
